src/wallet/rpcwallet.cpp

```diff
@@ -1164,6 +1175,10 @@ UniValue ListReceived(const UniValue& params, bool fByAccounts)
             if (!ExtractDestination(txout.scriptPubKey, address))
                 continue;
 
+            if (fFilterAddress && !(filterAddress == address)) {
```

Comment:
It's not a `CNoDestination`, it's a `CTxDestination`, which is a boost variant. Some supported versions of boost do not support `operator!=` for variants.